### ISSUE TITLE
node: seed mode should have a different concrete struct

### ISSUE THREAD

**Issue by cmwaters** [tendermint/tendermint#6775]
## Summary

Currently, a seed node uses the same struct as a full node or validator. However, seed nodes use such a small fraction of the services that I don't believe justifies using the same struct. Instead it causes the methods to have numerous conditional statements (if `mode != seed` then do this) and leaves many fields as `nil` which could have unknown consequences. Since these concrete types just need to fulfil a simple `Service` interface (start, stop), I propose that seed have it's own struct.

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by alexanderbez** [comment]
ACK

**Comment by marbar3778** [comment]
LGTM

**Comment by williambanfield** [comment]
Strongly agree with this!

**Comment by faddat** [comment]
Can someone please eli5 for me:

what is different between a seed node and a full node?

thank you

**Comment by alexanderbez** [comment]
@faddat, the Tl;DR is the idea that a seed node runs the PEX reactor only, and thus is responsible for peer exchange only to support a strongly connected p2p topology. A full node, on the other hands, is a node that runs all the reactors, i.e it can participate in consensus, serve query requests, etc...

**Comment by cmwaters** [comment]
Hey @spaceCh1mp, we're most likely planning on deprecating the seed node in v0.37 when we switch to libp2p so this work isn't of high importance. I see, however, that you've already opened a PR and given that it's a relatively small change that improves the code hygiene, I don't think it will be too much of a problem to review and merge.

**Comment by spaceCh1mp** [comment]
Oh alright, that's fine. This was light work. I'll look around to see if there are other issues I could take a go at. Thank you
